cmd/derper/derper.go

```diff
@@ -185,6 +185,23 @@ func main() {
 	mux.HandleFunc("/bootstrap-dns", handleBootstrapDNS)
 	mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This "/" here doesn't mean only /. It means anything not covered by a more specific handler (like a mount point)

By moving it here, this is no longer on the actual HTML pages we serve (over Tailscale) via the line 226 (tsweb.Debugger) registration.

The two endpoints I want to not send these headers on are /derp and /generate_204 (the two that tailscaled hit).

So you can move it back where you add it as long as you omit those path prefixes.

Or add all this stuff to some common place (new func in tsweb, like `tsweb.AddVariousHeadersToAppeaseSecurityScanners(w http.ResponseWriter)`?) and then use it here and in the tsweb debug handlers too? Your call.

But by making the func in tailscale/tailscale, we can probably remove duplication in our other repos too.

Reply:
I was thinking of creating a helper function in other repos, but I suppose we import tsweb everywhere so this makes more sense.
Added some helpers in tsweb and sprinkled them around derper mux.